apps/src/lab2/lab2Redux.ts

```diff
@@ -47,6 +45,7 @@ import ProjectManager from './projects/ProjectManager';
 import ProjectManagerFactory from './projects/ProjectManagerFactory';
 import {getPredictResponse} from './projects/userLevelsApi';
 import {setProjectTooLarge} from './redux/lab2ProjectRedux';
+import {isReadOnlyWorkspace} from './redux/lab2ReduxSelectors';
```

Comment:
Just curious about the refactor reason - I see `isReadOnlyWorkspace` is already being used in `setProjectAndLevelData`. Are you planning to use the selector in a different place and why will it cause a circular dependency now? Thanks!